components/table/table-wrapper.js

```diff
@@ -575,12 +576,26 @@ export class TableWrapper extends RtlMixin(PageableMixin(SelectionMixin(LitEleme
 	}
 
 	async _handleTableChange() {
-		await new Promise(resolve => requestAnimationFrame(resolve));
+		const update = async() => {
+			await new Promise(resolve => requestAnimationFrame(resolve));
 
-		this._updateItemShowingCount();
-		this._applyClassNames();
-		this._syncColumnWidths();
-		this._updateStickyTops();
+			this._updateItemShowingCount();
+			this._applyClassNames();
+			this._syncColumnWidths();
+			this._updateStickyTops();
+			this.dispatchEvent(new CustomEvent('d2l-table-changed', { bubbles: true, composed: false }));
+		};
+		const debounceDelay = 300;
+		if (!this._tableChangeTimeout) {
+			await update();
+			this._tableChangeTimeout = setTimeout(() => this._tableChangeTimeout = null, debounceDelay);
+		} else {
+			clearTimeout(this._tableChangeTimeout);
```

Comment:
Does it ever fall into this block for our demos?

This is probably ok. One thing I would note though is that this logic does not defer the update until the 300ms timeout, but rather, it defers the update until there is a 300ms timeout without a change. If a change happens every 290ms indefinitely, we'd never update (aside from the first update).